### ISSUE TITLE
swTFreeze(..., eos='gsw') and gsw_t_freezing() not the same

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1488]
As the title says, even though `swTFreeze()` should just be calling `gsw_t_freezing()`. The reason for this is that the pressure argument for `gsw_t_freezing()` is fixed as `p=0` when passed through `swTFreeze()`, which it probably shouldn't be (since `swTFreeze()` has a `pressure=` argument).

E.g. line [808](https://github.com/dankelley/oce/blob/73d602857c3f4790da3eb86c96363c2b4bf12bf2/R/sw.R#L808) in `sw.R` should be:

```r
        res <- gsw::gsw_t_freezing(SA=SA, p=pressure, saturation_fraction=saturation_fraction)
```

I didn't do this as a PR because I wasn't sure that the above is actually the *right* way to fix this, given the use of `lookWithin()` to extract other relevant fields. And actually looking at the code makes me think that the `pressure=` argument isn't even being used in the function. So, I wanted @dankelley to take a look.

**Comment by richardsc** [comment]
I think this is still wrong, or at least *inconsistent*, because of the handling of the `pressure=` arg in the `swTFreeze()` call vs the use of `lookWithin()` to get stuff out of the CTD object. In any case, I would expect the below example to give very similar TFreeze, but it doesn't, because how the `pressure` arg is being passed seems to be different:
```r
library(oce)
data(ctd)
p <- ctd[['pressure']]
plot(swTFreeze(ctd), p, ylim=rev(range(p)), type='l')
lines(swTFreeze(ctd, eos='gsw'), p, col=2)
legend('bottomright', c('UNESCO', 'GSW'), lty=1, col=1:2)
```
![image](https://user-images.githubusercontent.com/233584/52232762-b6948100-2893-11e9-841c-f80b6352514e.png)

Perhaps, if @dankelley isn't seeing the obvious fix here, we can save it for a f2f chat. I think if I actually sit down and look at the code (for more than 1 minute) I can parse out the best course of action here ...

**Comment by dankelley** [comment]
Thanks. I figured it out, I think.  Please look at 30252ac20b18460697da2c6076ec8d89450b3fd0 and also I made a test code (slight variant of the @richardsc code) at https://github.com/dankelley/oce-issues/tree/master/14xx/1488 that yields a nice pic like below. Note that the diff is way under a millidegree, which jives with my hypothesis that gsw framed the function to mimic the unesco (which makes sense since I doubt that there were new measurements made in a high-pressure tank etc).

![1488a](https://user-images.githubusercontent.com/99469/52234638-813e6200-2898-11e9-8a72-4213f0209346.png)

**Comment by dankelley** [comment]
f2f done; my work plan is now
* [ ] doc lookWithin and put in namespace
* [ ] clean docs on swTFreeze (e.g. wrt in-situ, like on teos)
* [ ] email to CR the info on that unesco report on a whoi workshop

**Comment by dankelley** [comment]
I did the items in my checklist ("develop" branch commit c08bdd2e5377bdc3a0de8fd1d39505aed020d4f8). Of particular note is that now `examples(swTFreeze)` does the check value, including explaining the temperature scale change, and it also makes a graph showing that the unesco and gsw formulations are very similar for the North-Atlantic `data(ctd)` dataset.

I think @richardsc and I are now clear on what pressure means. I think the docs for `?swTFreeze` could use an extra paragraph on the multiple modes of usage (e.g. sometime potential temperature, etc ... it all depends on what a person is trying to calculate!) as CR and I spoke of in our f2f today, but I'm inclined to leave that with him. The main thing (for me) is that now the phrase in-situ appears at the start of the docs.

If I get that doc from the WHOI workshop, and if that sheds more light on the unesco stream of calculation, then I'll be inclined to add a bit more to the `?swTFreeze` docs.

PS. the `?lookWithin` doc has been revised a bit, but I find it quite hard to explain, so I added some examples.
